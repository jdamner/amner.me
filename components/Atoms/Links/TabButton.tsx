import React from "react";
/* API */
import { useReducedMotion } from "framer-motion";

/* Components */
import MotionButton from "./MotionButton";

export default function TabButton(
  props: React.DetailedHTMLProps<
    React.HTMLAttributes<HTMLButtonElement>,
    HTMLButtonElement
  > & { active?: boolean; delay?: number },
) {
  const { active, ...rest } = props;
  const classes = [
    "text-xl md:text-3xl font-bold mr-3 md:mr-0 group hover:underline",
    active
      ? "text-gray-900 dark:text-gray-300"
      : "text-gray-300 dark:text-gray-500",
  ];

  const newProps = {
    ...rest,
    className: `${rest.className} ${classes.join(" ")}`,
  };

  return useReducedMotion() ? (
    <button {...newProps} />
  ) : (
    <MotionButton {...newProps} />
  );
}
