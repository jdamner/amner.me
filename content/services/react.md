---
title: Javascript & React
---

I've worked with [React](https://reactjs.org) and [React Native](https://reactnative.dev) for a number of years and find the stateful nature of the React framework to really work well in my head. I love the component based approach and the ability to create reusable components that can be used in multiple places.

This site is build using React and the [NextJS](https://www.nextjs.org/) framework. I parse [markdown](https://daringfireball.net/projects/markdown/) files for the content. This site is totally open source too, so if you're curious about how it's built, you can check out the [source code](https://www.github.com/jdamner/amner.me).

Also, as a statically generated site, it's super fast and easy to deploy.
