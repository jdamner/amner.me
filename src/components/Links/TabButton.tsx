/* API */
import { useReducedMotion } from "framer-motion"

/* Components */
import MotionButton from "./MotionButton";

/**
 * Tab Button 
 * 
 * @param {JSX.IntrinsicElements['button'] & { active: boolean }} props 
 * @returns {JSX.Element}
 */
export default function TabButton(props: JSX.IntrinsicElements['button'] & { active: boolean } ): JSX.Element {

    const classes = [
        "text-xl md:text-3xl font-bold mr-3 md:mr-0 group hover:underline",
        props.active ? "text-gray-900 dark:text-gray-300" : "text-gray-300 dark:text-gray-500"
    ]
    props.className && classes.push(props.className)

    props = {
        ...props,
        className: classes.join(' '),
    }

    // remove `active` prop
    delete props.active
    return useReducedMotion() ? 
        <button { ...props } /> :
        <MotionButton { ...props } />
}
