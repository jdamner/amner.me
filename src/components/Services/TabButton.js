import React from "react";

export default function TabButton({ onClick, name, active, index, image }) {
    return (
        <div 
            className={`tab-button ${active ? "active" : ""}`}
            onClick={onClick} 
            onKeyDown={onClick} 
            role='button' 
            aria-current={active} 
            tabIndex={index}>
                {image ?? null}
            <h3>{ name }</h3>
        </div>
    );
}