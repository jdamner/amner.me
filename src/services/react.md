---
title: React
thumbnail: ../assets/react-logo.png
---

Working with SQL in raw queries, and using frameworks such as Eloquent and WP_Query.
Coaching and supporting a team of developers grow their skills and meet their objectives
Supporting Project Managers and Business Development teams for technical guidance on quoting and supporting project lifecycles.
Debugging and maintaining PHP projects built in many frameworks, including bespoke PHP frameworks developed in house.
Developing systems that interact with complex data, such as data provided via third-party tools and services that run on device and output in raw formats.