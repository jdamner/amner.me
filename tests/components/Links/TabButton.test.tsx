import { render } from "@testing-library/react"
import TabButton from "../../../src/components/Links/TabButton"

describe("TabButton", () => {
  it("renders correctly", () => {
    const handleClick = jest.fn()
    const { container } = render(<TabButton index={1} name={'Test'} onClick={handleClick} active={true} />)
    expect(container).toMatchSnapshot()
  })
})
