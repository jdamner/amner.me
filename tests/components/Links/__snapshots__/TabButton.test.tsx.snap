// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabButton renders correctly 1`] = `
<div>
  <button
    aria-current="true"
    class="text-xl md:text-3xl font-bold text-gray-900"
    tabindex="1"
  >
    Test
  </button>
</div>
`;
